cdisc_rules_engine/operations/library_model_column_order.py

```diff
@@ -0,0 +1,190 @@
+                    if v["name"].lower().endswith("test"):
+                        test_index = i
+                variables_metadata = (
+                    findings_class_variables[: test_index + 1]
+                    + variables_metadata
+                    + findings_class_variables[test_index + 1 :]
+                )
+
+            gen_obs_class_metadata: dict = self._get_model_class_metadata(
+                model_details, GENERAL_OBSERVATIONS_CLASS
+            )
+            identifiers_metadata, timing_metadata = self._sort_class_variables_by_role(
+                gen_obs_class_metadata["classVariables"]
+            )
+            # Identifiers are added to the beginning and Timing to the end
+            if identifiers_metadata:
+                identifiers_metadata.sort(key=lambda item: item["ordinal"])
+                variables_metadata = identifiers_metadata + variables_metadata
+            if timing_metadata:
+                timing_metadata.sort(key=lambda item: item["ordinal"])
+                variables_metadata = variables_metadata + timing_metadata
+
+        return variables_metadata
+
+    def _get_model_type_and_version(self, model_link) -> Tuple:
+        link = model_link.get("href")
+        if "sdtm" in link:
+            model_type = "sdtm"
+            model_version = link.split("/")[-1]
+        else:
+            # TODO expand to support CDASH and ADAM
+            model_type = ""
+            model_version = ""
+        return model_type, model_version
+
+    def _get_model_class_metadata(
+        self,
+        model_details: dict,
+        class_title: str,
+    ) -> dict:
+        """
+        Extracts metadata of a certain class
+        from given standard model details.
+        """
+        class_metadata: Optional[dict] = search_in_list_of_dicts(
+            model_details.get("classes", []),
+            lambda item: item["label"] == class_title,
+        )
+        if not class_metadata:
+            raise ValueError(
+                f"Model class metadata is not found in CDISC Library. "
+                f"standard={self.params.standard}, "
+                f"version={self.params.standard_version}, "
+                f"class={class_title}"
+            )
+        return class_metadata
+
+    def _get_model_domain_metadata(self, model_details, domain_name) -> Tuple:
+        # Get domain metadata from model
+        domain_details: Optional[dict] = search_in_list_of_dicts(
+            model_details.get("datasets", []), lambda item: item["name"] == domain_name
+        )
+
+        if not domain_details:
+            raise ValueError(
+                f"Model domain metadata is not found in CDISC Library. "
+                f"standard={self.params.standard}, "
+                f"version={self.params.standard_version}, "
+                f"domain={domain_name}"
+            )
+
+        return domain_details
+
+    def _sort_class_variables_by_role(
```

Comment:
I would rename this to "_group_class_variables_by_role"

Reply:
updated.